store/storage/pebbledb/db.go

```diff
@@ -161,13 +196,84 @@ func (db *Database) ApplyChangeset(version uint64, cs *store.Changeset) error {
 	return b.Write()
 }
 
-// Prune for the PebbleDB SS backend is currently not supported. It seems the only
-// reliable way to prune is to iterate over the desired domain and either manually
-// tombstone or delete. Either way, the operation would be timely.
+// Prune removes all versions of all keys that are <= the given version.
+//
+// Note, the implementation of this method is inefficient and can be potentially
```

Comment:
is this synchronous?

Reply:
The method itself is, yes. But the caller can call it in a goroutine.